pkg/systemd/logs.jsx

```diff
@@ -0,0 +1,429 @@
+    const [showTextSearch, setShowTextSearch] = useState(false);
+    const [textFilter, setTextFilter] = useState(full_grep);
+    const [timeFilter, setTimeFilter] = useState(getTimeFilterOption(options));
+    const [updateIdentifiersList, setUpdateIdentifiersList] = useState(true);
+
+    useEffect(() => {
+        checkJournalctlGrep(setShowTextSearch);
+
+        function onNavigate() {
+            const { options, path } = cockpit.location;
+            const full_grep = getGrepFiltersFromOptions({ options })[0];
+
+            if (path.length == 1) return;
+
+            setJournalPrio(getPrioFilterOption(options));
+            setIdentifiersFilter(options.tag || _("All"));
+            setTextFilter(full_grep);
+            setTimeFilter(getTimeFilterOption(options));
+        }
+
+        cockpit.addEventListener("locationchanged", onNavigate);
+        return () => cockpit.removeEventListener("locationchanged", onNavigate);
+    }, []);
+
+    if (path.length == 1) {
+        return <LogEntry />;
+    } else if (path.length > 1) { /* redirect */
+        console.warn("not a journal location: " + path);
+        cockpit.location = '';
+    }
+
+    const updateUrl = (options) => {
+        cockpit.location.go([], options);
+    };
+
+    const onJournalPrioChange = (value) => {
+        setUpdateIdentifiersList(false);
+
+        updateUrl(Object.assign(options, { priority: value }));
+    };
+
+    const onIdentifiersFilterChange = (value) => {
+        setUpdateIdentifiersList(false);
+
+        if (value == _("All")) {
+            delete options.tag;
+            updateUrl(Object.assign(options));
+        } else {
+            updateUrl(Object.assign(options, { tag: value }));
+        }
+    };
+
+    const onTextFilterChange = (value) => {
+        setUpdateIdentifiersList(true);
+
+        updateUrl(Object.assign(getOptionsFromTextInput(value)));
+    };
+
+    const onTimeFilterChange = (newTimeFilter) => {
+        setUpdateIdentifiersList(true);
+
+        if (newTimeFilter.key == 'boot' && newTimeFilter.value !== "0") // Don't follow if specific boot is picked
+            setDataFollowing(false);
+        else if (options.boot && options.boot !== "0" && newTimeFilter.key !== "boot") // Start following is specific boot is removed
+            setDataFollowing(true);
+
+        // Remove all parameters which can be set up using filters
+        delete options.boot;
+        delete options.since;
+
+        cockpit.location.go([], Object.assign(options, { [newTimeFilter.key]: newTimeFilter.value }));
+    };
+
+    return (
+        <Page>
+            <PageSection id="journal" padding={{ default: 'noPadding' }}>
+                <Toolbar>
+                    <ToolbarContent>
+                        <ToolbarToggleGroup toggleIcon={<><span className="pf-c-button__icon pf-m-start"><FilterIcon /></span>{_("Toggle filters")}</>} breakpoint="md">
```

Comment:
When inserting some text in the searchinput, it grows and wraps to next line.
![Screen Shot 2021-08-10 at 12 35 08](https://user-images.githubusercontent.com/14921356/128856041-d4099d60-6601-4fac-94b4-97b2e56fe4e2.png)
![Screen Shot 2021-08-10 at 12 33 02](https://user-images.githubusercontent.com/14921356/128856050-1b455176-cebb-41c2-90ab-d1d27f091b4c.png)

Reply:
I cannot reproduce this. Does this happen on some specific browser, some specific width...?
I tried chrome and firefox, different width, typed very long strings and it never wrapped.

Reply:
Can't reproduce that on Firefox here, but the styling looks off: 
![Screenshot from 2021-08-11 11-12-55](https://user-images.githubusercontent.com/36541154/129003101-39393295-1cf8-4ad1-8a68-6b5590d5f75f.png)

In particular, the [X] having a blue underline seems wrong.